Create a universal2 binary for macOS

With Macs available with either the x86_64 and arm64 platforms (Related:#494), Apple has a separate binary type (called `universal2`) that provides a binary that will work on both platforms.

This _should_ be usable on Pyinstaller with the flag `--target-arch universal2` but does not seem to work with Pyinstaller 5.0.dev0 (from source) with the bootloader built for arm64 (see https://github.com/machawk1/wail/issues/494#issuecomment-865257873 )

The complaint is as follows but might stem on some dependencies also supporting this hybrid binary format:

```sh
3595 INFO: checking PKG
3595 INFO: Building PKG because PKG-00.toc is non existent
3595 INFO: Building PKG (CArchive) PKG-00.pkg
Traceback (most recent call last):
  File "/opt/homebrew/bin/pyinstaller", line 8, in <module>
    sys.exit(run())
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 126, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 758, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 705, in build
    exec(code, spec_namespace)
  File "/private/tmp/wail/WAIL.spec", line 23, in <module>
    exe = EXE(pyz,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 509, in __init__
    self.pkg = PKG(self.toc, cdict=kwargs.get('cdict', None),
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 208, in __init__
    self.__postinit__()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/datastruct.py", line 159, in __postinit__
    self.assemble()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 274, in assemble
    fnm = checkCache(fnm, strip=self.strip_binaries,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/utils.py", line 386, in checkCache
    osxutils.binary_to_target_arch(cachedfile, target_arch,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 332, in binary_to_target_arch
    assert target_arch != 'universal2', \
AssertionError: /opt/homebrew/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/_struct.cpython-39-darwin.so is not a fat binary!
```

This same behavior occurs with a dummy program:

```sh
machawk1@Mats-M1-Mac /tmp % echo "print('hello world')" > test.py
machawk1@Mats-M1-Mac /tmp % pyinstaller --onefile --target-arch universal2 hello.py
41 INFO: PyInstaller: 5.0.dev0
41 INFO: Python: 3.9.5
47 INFO: Platform: macOS-11.4-arm64-arm-64bit
47 INFO: wrote /private/tmp/hello.spec
49 INFO: UPX is not available.
49 INFO: Extending PYTHONPATH with paths
['/private/tmp', '/private/tmp']
55 INFO: checking Analysis
61 INFO: checking PYZ
62 INFO: EXE target arch: universal2
62 INFO: Code signing identity: None
62 INFO: checking PKG
62 INFO: Building PKG because PKG-00.toc is non existent
62 INFO: Building PKG (CArchive) PKG-00.pkg
Traceback (most recent call last):
  File "/opt/homebrew/bin/pyinstaller", line 8, in <module>
    sys.exit(run())
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 126, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 758, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 705, in build
    exec(code, spec_namespace)
  File "/private/tmp/hello.spec", line 23, in <module>
    exe = EXE(pyz,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 509, in __init__
    self.pkg = PKG(self.toc, cdict=kwargs.get('cdict', None),
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 208, in __init__
    self.__postinit__()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/datastruct.py", line 159, in __postinit__
    self.assemble()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 274, in assemble
    fnm = checkCache(fnm, strip=self.strip_binaries,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/utils.py", line 386, in checkCache
    osxutils.binary_to_target_arch(cachedfile, target_arch,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 332, in binary_to_target_arch
    assert target_arch != 'universal2', \
AssertionError: /opt/homebrew/Cellar/python@3.9/3.9.5/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/zlib.cpython-39-darwin.so is not a fat binary!
machawk1@Mats-M1-Mac /tmp % 
```

I am still unable to build a universal2 binary with the latest release, Pyinstaller 4.5.1.

```sh
/tmp % echo "print('hello world')" > hello.py
/tmp % pyinstaller --onefile --target-arch universal2 hello.py
24 INFO: PyInstaller: 4.5.1
24 INFO: Python: 3.9.6
29 INFO: Platform: macOS-11.5.2-arm64-arm-64bit
29 INFO: wrote /private/tmp/hello.spec
30 INFO: UPX is not available.
31 INFO: Extending PYTHONPATH with paths
['/private/tmp', '/private/tmp']
119 INFO: checking Analysis
121 INFO: Building because /private/tmp/hello.py changed
121 INFO: Initializing module dependency graph...
121 INFO: Caching module graph hooks...
124 INFO: Analyzing base_library.zip ...
1312 INFO: Processing pre-find module path hook distutils from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks/pre_find_module_path/hook-distutils.py'.
1312 INFO: distutils: retargeting to non-venv dir '/opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/lib/python3.9'
2285 INFO: Caching module dependency graph...
2351 INFO: running Analysis Analysis-00.toc
2353 INFO: Analyzing /private/tmp/hello.py
2353 INFO: Processing module hooks...
2353 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2354 INFO: Loading module hook 'hook-lib2to3.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2360 INFO: Loading module hook 'hook-encodings.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2388 INFO: Loading module hook 'hook-distutils.util.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2389 INFO: Loading module hook 'hook-pickle.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2390 INFO: Loading module hook 'hook-heapq.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2390 INFO: Loading module hook 'hook-difflib.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2391 INFO: Loading module hook 'hook-multiprocessing.util.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2391 INFO: Loading module hook 'hook-sysconfig.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2394 INFO: Loading module hook 'hook-xml.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2418 INFO: Loading module hook 'hook-distutils.py' from '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks'...
2426 INFO: Looking for ctypes DLLs
2433 INFO: Analyzing run-time hooks ...
2434 INFO: Including run-time hook '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pkgutil.py'
2435 INFO: Including run-time hook '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_multiprocessing.py'
2436 INFO: Including run-time hook '/opt/homebrew/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_inspect.py'
2439 INFO: Looking for dynamic libraries
2465 INFO: Looking for eggs
2465 INFO: Using Python library /opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/Python
2466 INFO: Warnings written to /private/tmp/build/hello/warn-hello.txt
2478 INFO: Graph cross-reference written to /private/tmp/build/hello/xref-hello.html
2482 INFO: checking PYZ
2483 INFO: Building because toc changed
2483 INFO: Building PYZ (ZlibArchive) /private/tmp/build/hello/PYZ-00.pyz
2688 INFO: Building PYZ (ZlibArchive) /private/tmp/build/hello/PYZ-00.pyz completed successfully.
2691 INFO: EXE target arch: universal2
2691 INFO: Code signing identity: None
2691 INFO: checking PKG
2691 INFO: Building PKG because PKG-00.toc is non existent
2691 INFO: Building PKG (CArchive) PKG-00.pkg
Traceback (most recent call last):
  File "/opt/homebrew/bin/pyinstaller", line 8, in <module>
    sys.exit(run())
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 126, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 815, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 762, in build
    exec(code, spec_namespace)
  File "/private/tmp/hello.spec", line 23, in <module>
    exe = EXE(pyz,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 509, in __init__
    self.pkg = PKG(self.toc, cdict=kwargs.get('cdict', None),
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 208, in __init__
    self.__postinit__()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/datastruct.py", line 159, in __postinit__
    self.assemble()
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/api.py", line 274, in assemble
    fnm = checkCache(fnm, strip=self.strip_binaries,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/building/utils.py", line 387, in checkCache
    osxutils.binary_to_target_arch(cachedfile, target_arch,
  File "/opt/homebrew/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 334, in binary_to_target_arch
    assert target_arch != 'universal2', \
AssertionError: /opt/homebrew/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/Python is not a fat binary!
```

With the cross-archtecture version of Python installed, attempting to compile still results in errors of dependencies not being available for the `universal2` platform.

```py
Requirement already satisfied: wxPython in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (4.1.1)
Requirement already satisfied: six in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from wxPython) (1.16.0)
Requirement already satisfied: numpy in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from wxPython) (1.21.1)
Requirement already satisfied: pillow in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from wxPython) (8.3.1)
WARNING: You are using pip version 21.1.3; however, version 21.2.3 is available.
You should consider upgrading via the '/Library/Frameworks/Python.framework/Versions/3.9/bin/python3 -m pip install --upgrade pip' command.
Requirement already satisfied: biplist in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 1)) (1.0.3)
Requirement already satisfied: black in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 2)) (21.7b0)
Requirement already satisfied: dmgbuild in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 3)) (1.5.2)
Requirement already satisfied: ds-store in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 4)) (1.3.0)
Requirement already satisfied: mac-alias in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 5)) (2.2.0)
Requirement already satisfied: pathlib2 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 6)) (2.3.6)
Requirement already satisfied: pyinstaller>=4.4 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 7)) (4.5.1)
Requirement already satisfied: pypubsub in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 8)) (4.0.3)
Requirement already satisfied: pystache in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 9)) (0.5.4)
Requirement already satisfied: requests in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 10)) (2.26.0)
Requirement already satisfied: schedule in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 11)) (1.1.0)
Requirement already satisfied: six in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 12)) (1.16.0)
Requirement already satisfied: wheel in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 13)) (0.37.0)
Requirement already satisfied: wxPython in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from -r requirements.txt (line 14)) (4.1.1)
Requirement already satisfied: altgraph in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from pyinstaller>=4.4->-r requirements.txt (line 7)) (0.17)
Requirement already satisfied: macholib>=1.8 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from pyinstaller>=4.4->-r requirements.txt (line 7)) (1.14)
Requirement already satisfied: setuptools in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from pyinstaller>=4.4->-r requirements.txt (line 7)) (56.0.0)
Requirement already satisfied: pyinstaller-hooks-contrib>=2020.6 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from pyinstaller>=4.4->-r requirements.txt (line 7)) (2021.2)
Requirement already satisfied: mypy-extensions>=0.4.3 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (0.4.3)
Requirement already satisfied: pathspec<1,>=0.8.1 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (0.9.0)
Requirement already satisfied: appdirs in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (1.4.4)
Requirement already satisfied: regex>=2020.1.8 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (2021.8.3)
Requirement already satisfied: tomli<2.0.0,>=0.2.6 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (1.2.1)
Requirement already satisfied: click>=7.1.2 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from black->-r requirements.txt (line 2)) (8.0.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from requests->-r requirements.txt (line 10)) (1.26.6)
Requirement already satisfied: certifi>=2017.4.17 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from requests->-r requirements.txt (line 10)) (2021.5.30)
Requirement already satisfied: idna<4,>=2.5 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from requests->-r requirements.txt (line 10)) (3.2)
Requirement already satisfied: charset-normalizer~=2.0.0 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from requests->-r requirements.txt (line 10)) (2.0.4)
Requirement already satisfied: pillow in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from wxPython->-r requirements.txt (line 14)) (8.3.1)
Requirement already satisfied: numpy in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from wxPython->-r requirements.txt (line 14)) (1.21.1)
WARNING: You are using pip version 21.1.3; however, version 21.2.3 is available.
You should consider upgrading via the '/Library/Frameworks/Python.framework/Versions/3.9/bin/python3 -m pip install --upgrade pip' command.
Creating binary
/Library/Frameworks/Python.framework/Versions/3.9/bin/pyinstaller
/Library/Frameworks/Python.framework/Versions/3.9/bin/python3
23 INFO: PyInstaller: 4.5.1
23 INFO: Python: 3.9.6
30 INFO: Platform: macOS-11.5.2-arm64-arm-64bit
30 INFO: wrote /private/tmp/wail/WAIL.spec
32 INFO: UPX is not available.
33 INFO: Removing temporary files and cleaning cache in /Users/machawk1/Library/Application Support/pyinstaller
36 INFO: Extending PYTHONPATH with paths
['/private/tmp/wail', '/private/tmp/wail/bundledApps', '/private/tmp/wail']
104 INFO: checking Analysis
104 INFO: Building Analysis because Analysis-00.toc is non existent
104 INFO: Initializing module dependency graph...
105 INFO: Caching module graph hooks...
108 INFO: Analyzing base_library.zip ...
1365 INFO: Processing pre-find module path hook distutils from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks/pre_find_module_path/hook-distutils.py'.
1366 INFO: distutils: retargeting to non-venv dir '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9'
2324 INFO: Caching module dependency graph...
2387 INFO: running Analysis Analysis-00.toc
2391 INFO: Analyzing /private/tmp/wail/bundledApps/WAIL.py
2491 INFO: Processing pre-safe import module hook urllib3.packages.six.moves from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks/pre_safe_import_module/hook-urllib3.packages.six.moves.py'.
2940 INFO: Processing pre-safe import module hook win32com from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/pre_safe_import_module/hook-win32com.py'.
2989 INFO: Processing module hooks...
2989 INFO: Loading module hook 'hook-certifi.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/stdhooks'...
2991 INFO: Loading module hook 'hook-pubsub.core.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/stdhooks'...
2992 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
2992 INFO: Loading module hook 'hook-lib2to3.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3004 INFO: Loading module hook 'hook-_tkinter.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3053 INFO: checking Tree
3053 INFO: Building Tree because Tree-00.toc is non existent
3053 INFO: Building Tree Tree-00.toc
3057 INFO: checking Tree
3057 INFO: Building Tree because Tree-01.toc is non existent
3057 INFO: Building Tree Tree-01.toc
3086 INFO: checking Tree
3087 INFO: Building Tree because Tree-02.toc is non existent
3087 INFO: Building Tree Tree-02.toc
3088 INFO: Loading module hook 'hook-encodings.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3116 INFO: Loading module hook 'hook-distutils.util.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3117 INFO: Loading module hook 'hook-pickle.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3118 INFO: Loading module hook 'hook-heapq.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3118 INFO: Loading module hook 'hook-difflib.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3119 INFO: Loading module hook 'hook-multiprocessing.util.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3119 INFO: Loading module hook 'hook-sysconfig.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3123 INFO: Loading module hook 'hook-xml.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3144 INFO: Loading module hook 'hook-distutils.py' from '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks'...
3152 INFO: Looking for ctypes DLLs
3159 INFO: Analyzing run-time hooks ...
3161 INFO: Including run-time hook '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pkgutil.py'
3162 INFO: Including run-time hook '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_multiprocessing.py'
3163 INFO: Including run-time hook '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/hooks/rthooks/pyi_rth_inspect.py'
3163 INFO: Including run-time hook '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/rthooks/pyi_rth_certifi.py'
3166 INFO: Looking for dynamic libraries
3236 INFO: Looking for eggs
3236 INFO: Using Python library /Library/Frameworks/Python.framework/Versions/3.9/Python
3238 INFO: Warnings written to /private/tmp/wail/build/WAIL/warn-WAIL.txt
3252 INFO: Graph cross-reference written to /private/tmp/wail/build/WAIL/xref-WAIL.html
3257 INFO: checking PYZ
3257 INFO: Building PYZ because PYZ-00.toc is non existent
3257 INFO: Building PYZ (ZlibArchive) /private/tmp/wail/build/WAIL/PYZ-00.pyz
3534 INFO: Building PYZ (ZlibArchive) /private/tmp/wail/build/WAIL/PYZ-00.pyz completed successfully.
3538 INFO: EXE target arch: universal2
3538 INFO: Code signing identity: None
3538 INFO: checking PKG
3538 INFO: Building PKG because PKG-00.toc is non existent
3538 INFO: Building PKG (CArchive) PKG-00.pkg
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/bin/pyinstaller", line 8, in <module>
    sys.exit(run())
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/__main__.py", line 126, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 815, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 762, in build
    exec(code, spec_namespace)
  File "/private/tmp/wail/WAIL.spec", line 23, in <module>
    exe = EXE(pyz,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/api.py", line 509, in __init__
    self.pkg = PKG(self.toc, cdict=kwargs.get('cdict', None),
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/api.py", line 208, in __init__
    self.__postinit__()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/datastruct.py", line 159, in __postinit__
    self.assemble()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/api.py", line 274, in assemble
    fnm = checkCache(fnm, strip=self.strip_binaries,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/building/utils.py", line 387, in checkCache
    osxutils.binary_to_target_arch(cachedfile, target_arch,
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/PyInstaller/utils/osx.py", line 334, in binary_to_target_arch
    assert target_arch != 'universal2', \
AssertionError: /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/wx/_adv.cpython-39-darwin.so is not a fat binary!
```

--- a/bundledApps/MAKEFILE.sh
+++ b/bundledApps/MAKEFILE.sh
@@ -86,7 +86,7 @@ createBinary ()
 {
   echo "Creating binary"
   which pyinstaller
-  pyinstaller -p bundledApps ./bundledApps/WAIL.py --onefile --windowed --clean --icon="./build/icons/wail_blue.icns"
+  pyinstaller -p bundledApps ./bundledApps/WAIL.py --onefile --windowed --clean --target-arch universal2 --icon="./build/icons/wail_blue.icns"
   # Replace default version and icon information from pyinstaller 
   cp ./build/Info.plist ./dist/WAIL.app/Contents/Info.plist
   # Copy the bundledApps and support directories to inside WAIL.app/

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,7 @@
 attrdict
 biplist
 black
+charset_normalizer<3.0
 dmgbuild
 ds-store
 mac-alias

